charts/spire/charts/spire-server/templates/configmap.yaml

```
@@ -1,3 +1,21 @@
+{{- range $type, $tvals := .Values.customPlugins }}
+{{-   if not (has $type (list "KeyManager" "NodeAttestor" "UpstreamAuthority" "Notifier")) }}
+{{-     fail (printf "Unknown plugin type specified: %s" $type) }}
+{{-   end }}
```

```suggestion
{{-   if (has $type (list "KeyManager" "NodeAttestor" "UpstreamAuthority" "Notifier")) }}
{{-     fail (printf "Please use the existing configurables for this plugin type: %s" $type) }}
{{-   end }}
```

I think this is wrong. For UpstreamAuthority and NodeAttestors and such we have already other configurables that we use in the configMap.

---

No, this suggestion is incorrect. It makes it impossible to use any 3rd party plugin. See the included example for something that needs to work.